contracts/cw-proposal-single/src/state.rs

```diff
@@ -128,27 +128,18 @@ pub fn get_deposit_msg(
     }
 }
 
-pub fn get_return_deposit_msg(proposal: &Proposal) -> StdResult<Vec<CosmosMsg>> {
-    match &proposal.deposit_info {
-        Some(info) => {
-            // If it is closed only issue a refund if we are refunding
-            // failed proposals. Otherwise, the proposal is open and
-            // expired.
-            if proposal.status == Status::Rejected && !info.refund_failed_proposals {
-                return Ok(vec![]);
-            }
-
-            let transfer_msg = WasmMsg::Execute {
-                contract_addr: info.token.to_string(),
-                funds: vec![],
-                msg: to_binary(&cw20::Cw20ExecuteMsg::Transfer {
-                    recipient: proposal.proposer.to_string(),
-                    amount: info.deposit,
-                })?,
-            };
-            let transfer_msg: CosmosMsg = transfer_msg.into();
-            Ok(vec![transfer_msg])
-        }
-        None => Ok(vec![]),
-    }
+pub fn get_return_deposit_msg(
+    deposit_info: &CheckedDepositInfo,
+    proposer: &Addr,
+) -> StdResult<Vec<CosmosMsg>> {
+    let transfer_msg = WasmMsg::Execute {
```

Comment:
Will error if we attempt to refund zero tokens. Just need a little check. Not your bug. :)
```suggestion
    if deposit_info.deposit.is_zero() {
        return Ok(vec![])
    }
    let transfer_msg = WasmMsg::Execute {
```